File: platforms/ios/devkit/Sources/NativeblocksDevkit/feature/live/presenter/LiveState.swift
internal struct LiveState {
    var notificationState: NotificationState

    static func initial() -> LiveState {
        return LiveState(notificationState: .none)
    }
}

internal enum NotificationState: String {
    case none = "NONE"
    case connect = "CONNECT"
    case connecting = "CONNECTING"
    case notConnect = "NOT_CONNECT"
    case authRequire = "AUTH_REQUIRE"
}

internal enum Action {
    case auth
    case start
    case connect
    case disconnect
    case logout
}

enum LiveServiceAction: String {
    case auth = "AUTH"
    case start = "START"
    case connect = "CONNECT"
    case disconnect = "DISCONNECT"
    case logout = "LOGOUT"
    case stop = "STOP"
    case notificationDeleted = "NOTIFICATION_DELETED"
}

File: platforms/android/devkit/devkit/src/main/java/io/nativeblocks/devkit/feature/live/presenter/LiveState.kt
package io.nativeblocks.devkit.feature.live.presenter

internal data class LiveState(val notificationState: NotificationState) {

    companion object {
        fun initial() = LiveState(
            notificationState = NotificationState.AUTH_REQUIRE
        )
    }
}

internal enum class NotificationState {
    CONNECT,
    CONNECTING,
    NOT_CONNECT,
    AUTH_REQUIRE
}

internal enum class Action {
    START,
    CONNECT,
    DISCONNECT,
    LOGOUT
}
